bw6/src/lib.rs

```diff
@@ -94,24 +105,37 @@ pub struct Proof<E: RegisterEvaluations, C: RegisterCommitments, AC: RegisterCom
 }
 
 pub type SimpleProof = Proof<AffineAdditionEvaluationsWithoutBitmask, PartialSumsCommitments, ()>;
-pub type PackedProof = Proof<SuccinctAccountableRegisterEvaluations, PartialSumsAndBitmaskCommitments, BitmaskPackingCommitments>;
+pub type PackedProof = Proof<
+    SuccinctAccountableRegisterEvaluations,
+    PartialSumsAndBitmaskCommitments,
+    BitmaskPackingCommitments,
+>;
 pub type CountingProof = Proof<CountingEvaluations, CountingCommitments, ()>;
 
-
 const H_X: Fr = MontFp!("0");
 const H_Y: Fr = MontFp!("1");
 fn point_in_g1_complement() -> ark_bls12_377::G1Affine {
     ark_bls12_377::G1Affine::new_unchecked(H_X, H_Y)
 }
 
-// TODO: switch to better hash to curve when available
-pub fn hash_to_curve<G: CurveGroup>(message: &[u8]) -> G {
-    use blake2::Digest;
-    use ark_std::rand::SeedableRng;
+pub fn hash_to_curve_g1(message: &[u8]) -> G1Projective {
+    let wb_to_curve_hasher = MapToCurveBasedHasher::<
+        Projective<G1Config>,
+        DefaultFieldHasher<Sha256>,
+        WBMap<G1Config>,
+    >::new(DST_G1)
+    .unwrap();
+    wb_to_curve_hasher.hash(message).unwrap().into()
+}
```

Comment:
Arkworks' H2C interface is kinda garbage btw.  I need to split up https://github.com/arkworks-rs/algebra/pull/643 so it can be merged.